cpp/src/arrow/testing/random.cc

```diff
@@ -430,6 +429,15 @@ std::shared_ptr<Array> RandomArrayGenerator::BinaryWithRepeats(
   return *strings->View(binary());
 }
 
+std::shared_ptr<Array> RandomArrayGenerator::StringView(int64_t size, int32_t min_length,
+                                                        int32_t max_length,
+                                                        double null_probability,
+                                                        int64_t alignment,
+                                                        MemoryPool* memory_pool) {
+  return GenerateBinaryArray<StringViewType, uint32_t>(
```

Comment:
Ah, missed updating this since the ML discussion on signed vs unsigned. I'll update to int32_t.

I agree that a variable buffer count would be a handy parameter, but maybe it should be added in another PR.